can-route-pushstate.js

```diff
@@ -159,41 +159,59 @@ canReflect.assign(PushstateObservable.prototype, {
 
 			// If link is within the same domain and descendant of `root`.
 			if (window.location.host === linksHost) {
-				var root = cleanRoot();
-
+				var root = cleanRoot(),
```

Comment:
Is it possible to pull out `anchorClickHandler` (or the important part of it) so it is easier to test directly?  If we can avoid creating iframe based tests, we should do that.

Reply:
Will do!

Reply:
@justinbmeyer I changed the function, can you please review?